File: Thread/app/src/main/java/com/example/thread/MainActivity.java
package com.example.thread;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {
    private static final String TAG =MainActivity.class.getSimpleName() ;
   private Button startThread ,stopThread;
    private  boolean StopLoop;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        startThread=(Button) findViewById(R.id.button);
        stopThread=(Button) findViewById(R.id.button2);
        Log.i(TAG,"Thread id :"+Thread.currentThread().getId());
        startThread.setOnClickListener(this);
        stopThread.setOnClickListener(this);

    }

    @Override
    public void onClick(View view) {
        switch (view.getId()){
            case R.id.button:
                StopLoop=true;
                new Thread(new Runnable() {
                    @Override
                    public void run() {
                while (StopLoop) {
                    Log.i(TAG, "Thread id in while loop:" + Thread.currentThread().getId());
                }}
                }).start();
                break;
            case R.id.button2:StopLoop=false;
            break;
        }

    }
}

File: EvTask1/app/src/main/java/com/example/evtask1/MainActivity.kt
package com.example.evtask1

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import java.util.Collections.swap

class MainActivity : AppCompatActivity(),View.OnClickListener{
    private var a = arrayListOf<String>("One","Two","Three","Four","Five")

     var btn1:Button? =null
    var btn2:Button?=null
    var btn3:Button? =null
   var btn4:Button? =null
   var btn5:Button?=null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        btn1 = findViewById(R.id.btn1)
        btn2 = findViewById(R.id.btn2)
        btn3 = findViewById(R.id.btn3)
        btn4 = findViewById(R.id.btn4)
        btn5 = findViewById(R.id.btn5)
        (btn1)?.setOnClickListener(this)
        (btn2)?.setOnClickListener(this)
        (btn3)?.setOnClickListener(this)
        (btn4)?.setOnClickListener(this)
        (btn5)?.setOnClickListener(this)

        btn1?.text = a.get(3)
        btn2?.text = a.get(4)
        btn3?.text = a.get(0)
        btn4?.text = a.get(2)
        btn5?.text = a.get(1)

    }

    override fun onClick(v: View?) {
    when (v?.id) {
            R.id.btn1 -> {
              swap()
            }
        R.id.btn2 -> {
              swap()
        }
        R.id.btn3 -> {
           swap()
        }
        R.id.btn4 -> {
           swap()
        }
        R.id.btn5 -> {
          swap()
        }
        }
    }

     fun swap() {
        when (a.indexOf(btn1?.text)) {
            0 -> btn1?.text = a[4]
            1 -> btn1?.text = a[0]
            2 -> btn1?.text = a[1]
            3 -> btn1?.text = a[2]
            4 -> btn1?.text = a[3]
        }
        when (a.indexOf(btn2?.text)) {
            0 -> btn2?.text = a[4]
            1 -> btn2?.text = a[0]
            2 -> btn2?.text = a[1]
            3 -> btn2?.text = a[2]
            4 -> btn2?.text = a[3]
        }
        when (a.indexOf(btn3?.text)) {
            0 -> btn3?.text = a[4]
            1 -> btn3?.text = a[0]
            2 -> btn3?.text = a[1]
            3 -> btn3?.text = a[2]
            4 -> btn3?.text = a[3]
        }
        when (a.indexOf(btn4?.text)) {
            0 -> btn4?.text = a[4]
            1 -> btn4?.text = a[0]
            2 -> btn4?.text = a[1]
            3 -> btn4?.text = a[2]
            4 -> btn4?.text = a[3]
        }
        when (a.indexOf(btn5?.text)) {
            0 -> btn5?.text = a[4]
            1 -> btn5?.text = a[0]
            2 -> btn5?.text = a[1]
            3 -> btn5?.text = a[2]
            4 -> btn5?.text = a[3]
        }
    }
}

